File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file288.f90
subroutine print_array(cz, cxm, cym, bza, ex, ey, hz)
implicit none

integer :: cz, cxm, cym
DATA_TYPE, dimension(cxm + 1, cym + 1, cz + 1) :: bza
DATA_TYPE, dimension(cxm + 1, cym + 1, cz + 1) :: ex
DATA_TYPE, dimension(cxm + 1, cym + 1, cz + 1) :: ey
DATA_TYPE, dimension(cxm + 1, cym + 1, cz + 1) :: hz
integer :: i, j, k

do i = 1, cz + 1
do j = 1, cym + 1
do k = 1, cxm + 1
write(0, DATA_PRINTF_MODIFIER) bza(k, j, i)
write(0, DATA_PRINTF_MODIFIER) ex(k, j, i)
write(0, DATA_PRINTF_MODIFIER) ey(k, j, i)
write(0, DATA_PRINTF_MODIFIER) hz(k, j, i)
if (mod(((i - 1) * cxm) + j - 1, 20) == 0) then
write(0, *)
end if
end do
end do
end do
write(0, *)
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file288.cpp
void print_array(int cz, int cxm, int cym, float bza[], float ex[], float ey[], float hz[]){
 for (int i = 0; i < cz+1; i++){
 for (int j = 0; j < cym+1; j++){
 for (int k = 0; k < cxm+1; k++){
 printf("%f", bza[k + (j * cxm) + (i * cym*cxm)]);
 printf("%f", ex[k + (j * cxm) + (i * cym*cxm)]);
 printf("%f", ey[k + (j * cxm) + (i * cym*cxm)]);
 printf("%f", hz[k + (j * cxm) + (i * cym*cxm)]);
 if ((((i * cxm) + j) % 20) == 0){
 printf("\n");
 }
 }
 }
 printf("\n");
 }
}
